src/js/tweet-parser.js

```diff
@@ -30,79 +92,238 @@ TweetParser.prototype = {
+  if (userCase === null) {
+    text = tweet.querySelector(textAlternativeSelector) ? tweet.querySelector(textAlternativeSelector).textContent : null;
+  }
+  else {
+    // Case when we are in a Tweet Page, with its responses
+    if (pageCase === "tweet") {
+      // The first one is the main Tweet
+      if (num === 0) {
+        text = tweet.querySelector(textSelectorTweetPageCase) ? tweet.querySelector(textSelectorTweetPageCase).textContent : null;
+      }
+      // The other cases are the responses tweets
+      else {
+        text = tweet.querySelector(textSelectorTweetPageResponsesCase) ? tweet.querySelector(textSelectorTweetPageResponsesCase).textContent : null;
+      }
+    }
+    // Case when we are in the Tweeter Home Page, or a User Page
+    else {
+      text = tweet.querySelector(textSelector) ? tweet.querySelector(textSelector).textContent : null;
+    }
+  }
+  
+  return {
     username: user,
     text: text,
     id: tweetid,
-    links: getLinks(tweet),
     domObject: tweet,
-    analyzed: isAnalyzed 
+    analyzed: false,
+    logo: false
   };
 
-  return mock;
-
 };
 
-const indexTweets = (callback) => {
+const treatNewTweet = (tweet, num, callback) => {
+  const tweetInfo = getTweetInfo(tweet, null);
 
-  tweetsList.forEach(tweet => {
+  if (tweetInfo.username) {
 
-    const tweetInfo = getTweetInfo(tweet);
+    idAssignator(tweet);
+    tweet.setAttribute(usernameAttribute, tweetInfo.username);
+    callback(tweetInfo);
 
-    if (tweetInfo.username) {
+  }
+};
 
-      idAssignator(tweet);
-      tweet.setAttribute(usernameAttribute, tweetInfo.username);
-      callback(tweetInfo);
+const indexTweets = (callback) => {
 
-    }
+  tweetsList.forEach((tweet, num) => {
+
+    treatNewTweet(tweet, num, callback);
 
   });
 
 };
 
-const twitterMainChangeCallback = (change, callback) => {
-
-  if (!!change.querySelector(tweetJQuerySelector)) {
+const mainChangeCallback = (newNode, callback) => {
+
+  let auxTweetNode = null;
+  let auxSectionNode = null;
+  let newTweetNode = null;
+
+  // we have to check if there was a new tweet, or a new section
+  // to check it we have to consider whick user case we are in (logged / not logged)
+  if (userCase === null) {
+    try {
+      // we check if the new node is a new tweet itself
+      if (newNode.matches(tweetAlternativeSelector)) {
+        newTweetNode = newNode;
+      }
+      else {
+        // we check if the new node contains a new tweet
+        auxTweetNode = newNode.querySelector(tweetAlternativeSelector);
+        if (auxTweetNode) {
+          newTweetNode = auxTweetNode;
+          auxTweetNode = true;
+        }
+      }
+      // we check if the new node is a new section itself
+      auxSectionNode = newNode.matches(sectionAlternativeSelector);
+      if (!auxSectionNode) {
+        // we check if the new node contains a new section
+        auxSectionNode = newNode.querySelector(sectionAlternativeSelector);
+      }
+    } catch (e) {}
```

Comment:
catching an error and not logging/showing the error actually happened?
I have a somewhat limited understanding of JavaScript but I would not recommend NOT showing an error was thrown and caught. Makes for harder debugging when the rrors occur

Reply:
Agree. It was to avoid a detected exception situation. I will raise an issue about it to try to avoid the catch block with better practice code, preventing the code execution if the situation can raise the exception.